lang/java/reef-examples/src/main/java/org/apache/reef/examples/hello/HelloReefAzBatch.java

```diff
@@ -74,9 +88,10 @@ public static void main(final String[] args) throws InjectionException, IOExcept
 
     final Configuration partialConfiguration = getEnvironmentConfiguration();
     final Injector injector = Tang.Factory.getTang().newInjector(partialConfiguration);
+    final String containerRegistryServer = injector.getNamedInstance(ContainerRegistryServer.class);
     final AzureBatchRuntimeConfigurationProvider runtimeConfigurationProvider =
         injector.getInstance(AzureBatchRuntimeConfigurationProvider.class);
-    final Configuration driverConfiguration = getDriverConfiguration();
+    final Configuration driverConfiguration = getDriverConfiguration(!StringUtils.isEmpty(containerRegistryServer));
```

Comment:
>!StringUtils.isEmpty(containerRegistryServer) [](start = 69, length = 45)

Instead of duplicating the ContainerRegistryProvider.IsValid logic, you should you the IsValid call directly. However, I don't believe you should be performing this check at all. Why not just blindly pass the port configuration?

Reply:
Moved the BindImplementation to AzureBatchDriverConfigurationProviderImpl class instead of client.

---
In reply to: [201087770](https://github.com/sharathmalladi/reef/pull/5#discussion_r201087770) [](ancestors = 201087770)